infrastructure/base/modules/beanstalk/main.tf

```diff
@@ -0,0 +1,182 @@
+#
+# Site Server Security Groups
+# SSH access to and from the world
+# HTTP(S) access from the world
+#
+resource "aws_security_group" "site_server_ssh_security_group" {
+  vpc_id      = var.vpc.id
+  name        = "${var.project}-${var.environment}-public-ssh-sg"
+  description = "Security group for SSH access to and from the world - ${var.project} ${var.environment}"
+
+  tags = merge(
+    {
+      Name = "EC2 SSH access SG"
+    },
+    var.tags
+  )
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+resource "aws_security_group_rule" "ssh_ingress" {
+  type              = "ingress"
+  from_port         = 22
+  to_port           = 22
+  protocol          = "tcp"
+  cidr_blocks       = ["0.0.0.0/0"]
+  security_group_id = aws_security_group.site_server_ssh_security_group.id
+}
+
+resource "aws_security_group_rule" "ssh_egress" {
+  type        = "egress"
+  from_port   = 22
+  to_port     = 22
+  protocol    = "tcp"
+  cidr_blocks = [var.vpc.cidr_block]
+
+  security_group_id = aws_security_group.site_server_ssh_security_group.id
+}
+
+# Create elastic beanstalk application
+
+resource "aws_elastic_beanstalk_application" "application" {
+  name = var.application_name
+}
+
+# Create elastic beanstalk Environment
+
+resource "aws_elastic_beanstalk_environment" "application_environment" {
+  name                   = var.application_environment
+  application            = aws_elastic_beanstalk_application.application.name
+  solution_stack_name    = var.solution_stack_name
+  tier                   = var.tier
+  wait_for_ready_timeout = "20m"
+
+  setting {
```

Comment:
improvement, if you feel like it: you can DRY this up using https://developer.hashicorp.com/terraform/language/expressions/dynamic-blocks and a "locals" array of config values